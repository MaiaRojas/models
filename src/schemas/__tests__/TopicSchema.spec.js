const mongoose = require('mongoose/browser');
const { TopicSchema } = require('../')(mongoose);
const babelTopicJson = require('./fixtures/topics/babel');

describe('TopicSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, TopicSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should pass sync validation ...', () => {
    const doc = new mongoose.Document({
      slug: 'foo',
      repo: 'foo/bar',
      path: '.',
      version: '1.0.0',
      parserVersion: '2.0.0',
      track: 'js',
      locale: 'es-ES',
      title: 'Foo',
      stats: {
        partCount: 0,
        unitCount: 0,
        exerciseCount: 0,
        durationString: '0min',
        duration: 0,
      },
    }, TopicSchema);
    expect(doc.validateSync()).toBe(undefined);
  });

  it('should validate existing topic json', () => {
    const babelTopic = new mongoose.Document(babelTopicJson, TopicSchema);
    return babelTopic.validate();
  });
});
