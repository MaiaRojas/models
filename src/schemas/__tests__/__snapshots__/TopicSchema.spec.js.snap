// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicSchema should fail validation when fields missing 1`] = `
Object {
  "locale": [ValidatorError: Path \`locale\` is required.],
  "parserVersion": [ValidatorError: Path \`parserVersion\` is required.],
  "path": [ValidatorError: Path \`path\` is required.],
  "repo": [ValidatorError: Path \`repo\` is required.],
  "slug": [ValidatorError: Path \`slug\` is required.],
  "stats.duration": [ValidatorError: Path \`stats.duration\` is required.],
  "stats.durationString": [ValidatorError: Path \`stats.durationString\` is required.],
  "stats.exerciseCount": [ValidatorError: Path \`stats.exerciseCount\` is required.],
  "stats.partCount": [ValidatorError: Path \`stats.partCount\` is required.],
  "stats.unitCount": [ValidatorError: Path \`stats.unitCount\` is required.],
  "title": [ValidatorError: Path \`title\` is required.],
  "track": [ValidatorError: Path \`track\` is required.],
  "version": [ValidatorError: Path \`version\` is required.],
}
`;
